src/app/clusters/time-synchronization-server/time-synchronization-server.cpp

```diff
@@ -771,6 +771,13 @@ CHIP_ERROR TimeSynchronizationServer::GetDefaultNtp(MutableCharSpan & dntp)
 Span<TimeSyncDataProvider::TimeZoneStore> & TimeSynchronizationServer::GetTimeZone()
 {
     mTimeZoneObj.timeZoneList = mTimeZoneObj.timeZoneList.SubSpan(0, mTimeZoneObj.validSize);
+
+    // Make sure the sizes of the entries is correct
+    for (auto & tzStore : mTimeZoneObj.timeZoneList)
+    {
+        tzStore.timeZone.name = MakeOptional(CharSpan(tzStore.name, strlen(tzStore.name)));
```

Comment:
Need to ensure `strnlen` is called since there is no mention of null termination in tzStore.name. This could lead to garbage.

Reply:
Done.